python/ray/serve/api.py

```diff
@@ -718,3 +719,30 @@ def my_deployment_function(request):
     """
     _request_context = ray.serve.context._serve_request_context.get()
     return _request_context.multiplexed_model_id
+
+
+@PublicAPI(stability="alpha")
+def status() -> ServeStatus:
+    """Get status of Serve on the cluster.
+
+    Includes status of all HTTP Proxies, all active applications, and
+    their deployments.
+
+    .. code-block:: python
+
+            @serve.deployment(num_replicas=2)
+            class MyDeployment:
+                pass
+
+            serve.run(MyDeployment.bind())
+            status = serve.status()
+            assert status.applications["default"].status == "RUNNING"
+    """
+
+    client = get_global_client(raise_if_no_controller_running=False)
+    if client is None:
+        # Serve has not started yet
```

Comment:
nit: we can add a warning here.

Reply:
Will it be strange to add a warning in a status API when nothing is wrong? I was thinking the user should understand Serve hasn't started if the list of http proxies and applications is empty. What do you think?